worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -29,7 +28,13 @@ class ConnectionProvider(props: Properties) {
   // Class for storing the ip address of the host, along with an expiration date
   private case class CachedIpWithExpiration(ip: String, expiration: Long)
   // Cache for ip address and its expiration for a host
-  private val cachedIpWithExpiration: LinkedHashMap[String, CachedIpWithExpiration] = new LinkedHashMap()
+  private val cachedIpWithExpiration: java.util.Map[String, CachedIpWithExpiration] = Collections.synchronizedMap(new LinkedHashMap[String, CachedIpWithExpiration]() {
+    // There is only 1 element to be stored in this list at a time, so we remove the oldest entries once we reach 2+ entries
+    override def removeEldestEntry(eldest: java.util.Map.Entry[String, CachedIpWithExpiration]): Boolean = {
+      size > 1
```

Comment:
wait.... if we only have 1 entry, maybe it would be simpler just to have a variable for this (synchronized, or maybe @volatile?)

Reply:
Yeah probably simpler to do make is a single optional object. I think `@volatile` should be adequate over using `syncrhonized` since as long as there is a cached entry to use, we don't care about whether another thread tried to enter the same DNS result into the cache.

Reply:
On second thought, perhaps `synchronized` may be better here. It just makes the cache perform better if there isn't a race-condition between which IP value is set for the DNS cache